### PR TITLE
fix(index-handlers): fix bug with missing id in contributor affiliations #NP-45427

### PR DESCRIPTION
**Pull request by torbjokv**

### CODE REVIEW

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/UpdateIndexHandler.java [hunk 1]
```diff
@@ -35,30 +41,34 @@ public class UpdateIndexHandler implements RequestHandler<DynamodbEvent, Void> {
...
+    private final String dlqUrl;
 
     @JacocoGenerated
     public UpdateIndexHandler() {
-        this(new S3StorageReader(EXPANDED_RESOURCES_BUCKET), defaultOpenSearchClient(),
+        this(new S3StorageReader(new Environment().readEnv("EXPANDED_RESOURCES_BUCKET")), defaultOpenSearchClient(),
              new CandidateRepository(defaultDynamoClient()), new PeriodRepository(defaultDynamoClient()),
-             new NviCandidateIndexDocumentGenerator(defaultUriRetriver()));
+             new NviCandidateIndexDocumentGenerator(defaultUriRetriver(new Environment())), new NviQueueClient(),
+             new Environment());
```
Comment (hackon) [inline index-handlers/src/main/java/no/sikt/nva/nvi/index/UpdateIndexHandler.java:57]: You can extract new Environment() into constant

**Review by hackon — APPROVED** [review]

File: index-handlers/src/main/java/no/sikt/nva/nvi/index/UpdateIndexHandler.java [hunk 2]
```diff
@@ -120,10 +129,23 @@ private void updateIndex(DynamodbStreamRecord record) {
...
+
+            attempt(() -> queueClient.sendMessage(JsonUtils.dtoObjectMapper.writeValueAsString(Map.of(
+                "exception", getStackTrace(e),
+                "dynamodb", record
+            )), dlqUrl));
         }
     }
 
+    private static String getStackTrace(Exception e) {
+        var sw = new StringWriter();
```
Comment (brinxmat) [inline index-handlers/src/main/java/no/sikt/nva/nvi/index/UpdateIndexHandler.java:144]: sw -> stringWriter, boring, I know

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/UpdateIndexHandlerTest.java [hunk 3]
```diff
@@ -145,6 +158,19 @@ void shouldUpdateExistingIndexDocumentWhenIncomingEventIsModifyAndCandidateIsApp
         assertThat(document, is(equalTo(expectedDocument)));
     }
 
+    @Test
+    void indexingFailureShouldBeAddedToDlq()
+        throws JsonProcessingException {
+        var persistedCandidate = randomApplicableCandidate();
+        mockRepositories(persistedCandidate);
+        Optional<CandidateDao> failingCandidateLookup = Optional.empty();
```
Comment (brinxmat) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/UpdateIndexHandlerTest.java:166]: Just inline `Optional.<CandidateDao>empty()`

File: index-handlers/src/test/java/no/sikt/nva/nvi/index/UpdateIndexHandlerTest.java [hunk 4]
```diff
@@ -145,6 +158,19 @@ void shouldUpdateExistingIndexDocumentWhenIncomingEventIsModifyAndCandidateIsApp
         assertThat(document, is(equalTo(expectedDocument)));
     }
 
+    @Test
+    void indexingFailureShouldBeAddedToDlq()
+        throws JsonProcessingException {
+        var persistedCandidate = randomApplicableCandidate();
```
Comment (brinxmat) [inline index-handlers/src/test/java/no/sikt/nva/nvi/index/UpdateIndexHandlerTest.java:164]: Inline

File: nvi-commons/src/main/java/no/sikt/nva/nvi/common/queue/NviSendMessageResponse.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+        public Builder messageId(String messageId) {
+            this.builderMessageId = messageId;
+            return this;
+        }
+
+        public NviSendMessageResponse build() {
+            return new NviSendMessageResponse(builderMessageId);
+        }
+    }
```
Comment (brinxmat) [inline nvi-commons/src/main/java/no/sikt/nva/nvi/common/queue/NviSendMessageResponse.java:24]: Builder for a single field?

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**